core/templates/components/state-editor/state-hints-editor/state-hints-editor.component.ts

```diff
@@ -171,6 +171,7 @@ angular.module('oppia').component('stateHintsEditor', {
           StateHintsService.displayed = [];
           StateHintsService.saveDisplayedValue();
           ctrl.onSaveHints(StateHintsService.displayed);
+          $rootScope.$apply();
```

Comment:
Can you use `$applyAsync` instead?

Reply:
Done

Reply:
@srijanreddy98 can you please approve these changes, auto merging is block because of this.